mock/mockimpl/svcimpls/kvimplhello.go

```diff
@@ -15,18 +15,53 @@ func (x *kvImplHello) Register(h *hookHelper) {
 }
 
 func (x *kvImplHello) handleHelloRequest(source mock.KvClient, pak *memd.Packet) {
+	isInFeatureList := func(features []memd.HelloFeature, feature memd.HelloFeature) bool {
+		for _, foundFeature := range features {
+			if foundFeature == feature {
+				return true
+			}
+		}
+		return false
+	}
+
+	// TODO(brett19): This list should actually be moderated by the server
+	// version/features configured at the top level for the cluster.
+	availableFeatures := []memd.HelloFeature{
+		memd.FeatureDatatype,
+		memd.FeatureTCPNoDelay,
+		memd.FeatureSeqNo,
+		memd.FeatureTCPDelay,
+		memd.FeatureXattr,
+		memd.FeatureXerror,
+		memd.FeatureSelectBucket,
+		memd.FeatureSnappy,
+		memd.FeatureJSON,
+		memd.FeatureDuplex,
+		//memd.FeatureClusterMapNotif,
+		memd.FeatureUnorderedExec,
+		//memd.FeatureDurations,
+		//memd.FeatureAltRequests,
+		//memd.FeatureSyncReplication,
+		memd.FeatureCollections,
+		//memd.FeatureOpenTracing,
+		//memd.FeatureCreateAsDeleted,
```

Comment:
I think we do actually support this?

Reply:
We only halfway support it.  We pick up the flag and pass it through the subdoc executor, but we don't actually use it to force the created document to be deleted.